compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -138,6 +138,23 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 self.describe_place_with_options(moved_place, IncludingDowncast(true)),
             );
 
+            let reinit_spans = maybe_reinitialized_locations
+                .iter()
+                .take(3)
+                .map(|loc| {
+                    self.move_spans(self.move_data.move_paths[mpi].place.as_ref(), *loc)
+                        .args_or_use()
+                })
+                .collect::<Vec<Span>>();
+            if reinit_spans.len() == 1 {
+                err.span_label(reinit_spans[0], "this reinitialization might get skipped");
+            } else if reinit_spans.len() > 1 {
+                err.span_note(
+                    MultiSpan::from_spans(reinit_spans),
+                    "these reinitializations might get skipped",
+                );
+            }
```

Comment:
I love the new output, but I would like it if for the case where there are _more_ than 3 we provided something like

```
note: these 3 reinitializations and N others might get skipped
  --> $DIR/issue-83760.rs:30:9
   |
LL |         foo = Some(Struct);
   |         ^^^^^^^^^^^^^^^^^^
LL |     } else if true {
LL |         foo = Some(Struct);
   |         ^^^^^^^^^^^^^^^^^^
LL |     } else if true {
LL |         foo = Some(Struct);
   |         ^^^^^^^^^^^^^^^^^^
```

Reply:
Good idea.